cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -729,18 +729,9 @@ toTxOutInAnyEra era (TxOutAnyEra addr' val' mDatumHash refScriptFp) = do
   addr <- hoistEither $ toAddressInAnyEra era addr'
   val <- hoistEither $ toTxOutValueInAnyEra era val'
   (datum, refScript)
-    <- caseByronToMaryOrAlonzoEraOnwards
+    <- caseByronToAlonzoOrBabbageEraOnwards
          (const $ pure (TxOutDatumNone, ReferenceScriptNone))
-         (\w ->
-            caseAlonzoOnlyOrBabbageEraOnwards
-              (\wa ->
-                (,)
-                  <$> toTxAlonzoDatum (alonzoEraOnlyToAlonzoEraOnwards wa) mDatumHash
```

Comment:
Previously we could use datum-hashes from Alonzo era onwards.  If we don't support that use case, the code simplifies.